src/app/icd/ICDMonitoringTable.cpp

```diff
@@ -72,7 +72,7 @@ CHIP_ERROR ICDMonitoringEntry::Deserialize(TLV::TLVReader & reader)
             case to_underlying(Fields::kKey): {
                 ByteSpan buf(key.AsMutable<Crypto::Aes128KeyByteArray>());
                 ReturnErrorOnFailure(reader.Get(buf));
-                ReturnErrorOnFailure(this->SetKey(buf));
+                memcpy(key.AsMutable<Crypto::Aes128KeyByteArray>(), buf.data(), sizeof(Crypto::Aes128KeyByteArray));
```

Comment:
Why does calling `SetKey` not work as intended? With the comment, i think i'm missing the underlying issue.

Reply:
Whoa.  This is a major change that was made after reviews.....

Reply:
Yeah... enabling the PSA_CRYPTO test showcased this failure. But all CI should pass now.